python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -94,14 +94,20 @@ class AssembleCoaddConnections(pipeBase.PipelineTaskConnections,
     )
 
     def __init__(self, *, config=None):
-        # Override the connection's name template with config to replicate Gen2 behavior
-        config.connections.warpType = config.warpType
-        config.connections.warpTypeSuffix = makeCoaddSuffix(config.warpType)
+        super().__init__(config=config)
 
+        # Override the connection's name template with config to replicate Gen2 behavior
+        # This duplicates some of the logic in the base class, due to waning Gen2 and
+        # Gen3 configs to stay in sync. This should be removed when gen2 is deprecated
+        templateValues = {name: getattr(config.connections, name) for name in self.defaultTemplates.keys()}
```

Comment:
is `.keys()` needed here? If `self.defaultTemplates` is dict-like then it shouldn't be there. I have the same comment on the `self.allConnections.keys()` line below.